File: Adapter/PHP/CurrentMediaPlayer.php
<?php

require_once "IMediaPlayer.php";

class CurrentMediaPlayer implements IMediaPlayer{
    public function playMKV(string $filename) {
        echo "Playing file " . $filename . " with MKV format\n";
    }

    public function playMP4(string $filename) {
        echo "Playing file " . $filename . " with MP4 format\n";
    }
}

File: Adapter/Golang/CurrentMediaPlayer.go
package main

import "fmt"

type CurrentMediaPlayer struct {}

func (currentMediaPlayer *CurrentMediaPlayer) playMKV(filename string) {
    fmt.Println("Playing file", filename, "with MKV format")
}

func (currentMediaPlayer *CurrentMediaPlayer) playMP4(filename string) {
    fmt.Println("Playing file", filename, "with MKV format")
}